sdk/eventgrid/azure-messaging-eventgrid-namespaces/src/main/java/com/azure/messaging/eventgrid/namespaces/EventGridReceiverAsyncClient.java

```diff
@@ -385,12 +387,48 @@ public Mono<ReceiveResult> receive(Integer maxEvents, Duration maxWaitTime) {
             requestOptions.addQueryParam("maxEvents", String.valueOf(maxEvents), false);
         }
         if (maxWaitTime != null) {
+            // Ensure the http client timeout is longer than the maxWaitTime
+            addTimeoutToContext(requestOptions, maxWaitTime);
             requestOptions.addQueryParam("maxWaitTime", String.valueOf(maxWaitTime.getSeconds()), false);
         }
         return receiveWithResponse(topicName, subscriptionName, requestOptions).flatMap(FluxUtil::toMono)
             .map(protocolMethodData -> protocolMethodData.toObject(ReceiveResult.class));
     }
 
+    /**
+     * Receive a batch of Cloud Events from a subscription.
+     *
+     * @param maxEvents Max Events count to be received. Minimum value is 1, while maximum value is 100 events. If not
+     * specified, the default value is 1.
+     * @param maxWaitTime Max wait time value for receive operation in Seconds. It is the time in seconds that the
+     * server approximately waits for the availability of an event and responds to the request. If an event is
+     * available, the broker responds immediately to the client. Minimum value is 10 seconds, while maximum value is 120
+     * seconds. If not specified, the default value is 60 seconds.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return details of the Receive operation response on successful completion of {@link Mono} along with {@link Response}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<ReceiveResult>> receiveWithResponse(Integer maxEvents, Duration maxWaitTime,
+        RequestOptions requestOptions) {
+        // Generated convenience method for receiveWithResponse
+        if (maxEvents != null) {
+            requestOptions.addQueryParam("maxEvents", String.valueOf(maxEvents), false);
+        }
+        if (maxWaitTime != null) {
+            // Ensure the http client timeout is longer than the maxWaitTime
+            addTimeoutToContext(requestOptions, maxWaitTime);
```

Comment:
I chose this instead of considering the default as the user is expressing a desired timeout. If they said 5s and we waited 60s due to some other issue that feels wrong. Waiting 10s (with the buffer) still isn't perfect but seems better to me. Should I include in the javadoc that we set the HTTP timeout with the buffer? I didn't originally as it felt a bit too implementation focused.